### PR TITLE
Add support for Teresa yaml files per process type

### PR DESCRIPTION
**Pull request by aguerra**
By default teresa-processtype.yaml is used with teresa.yaml
as a fallback.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa/285)

### CODE REVIEW

File: pkg/server/deploy/config.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ import (
 )
 
 const (
+	ProcfileFileName       = "Procfile"
+	teresaYamlFileNameTmpl = "teresa%s%s.yaml"
```
Comment (yagonobre) [inline pkg/server/deploy/config.go]: It's not more exported?

Comment (yagonobre) [inline pkg/server/deploy/config.go]: Is it not more exported?

Comment (yagonobre) [inline pkg/server/deploy/config.go]: Nice :)

File: pkg/server/deploy/config.go [hunk 2]
```diff
@@ -85,28 +92,26 @@ func getDeployConfigFilesFromTarBall(tarBall io.ReadSeeker) (*DeployConfigFiles,
...
+				if err := deployFiles.fillTeresaYaml(tarReader); err != nil {
+					return nil, err
+				}
 			}
 		} else {
 			deployFiles.Procfile = make(map[string]string)
 			if err := readFileFromTarBall(tarReader, deployFiles.Procfile); err != nil {
 				return nil, err
 			}
 		}
```
Comment (drgarcia1986) [inline pkg/server/deploy/config.go]: Why you removed that last if? (`if deployFiles.TeresaYaml != nil && deployFiles.Procfile != nil`).
That "if" will prevent Teresa to read a bunch of files unnecessary, or am I miss something?

Comment (aguerra) [inline pkg/server/deploy/config.go]: Suppose that `deployFiles.TeresaYaml` is filled (not null) with `teresa.yaml`, `teresa-processtype.yaml` maybe the last one when iterating, so we can't break the loop. What do you think?

Comment (drgarcia1986) [inline pkg/server/deploy/config.go]: You have a point!

**Pull request merged**